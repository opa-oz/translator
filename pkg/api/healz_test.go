package api_test

import (
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/assert"
	"net/http"
	"net/http/httptest"
	"testing"
	"translator/pkg/api"

	"github.com/gin-gonic/gin"
)

func TestHealzRoute(t *testing.T) {
	r := gin.Default()

	r.GET("/healz", api.Healz)

	w := httptest.NewRecorder()
	req, _ := http.NewRequest(http.MethodGet, "/healz", nil)
	r.ServeHTTP(w, req)

	assert.Equal(t, http.StatusOK, w.Code)
	snaps.MatchSnapshot(t, w.Body.String())
}
