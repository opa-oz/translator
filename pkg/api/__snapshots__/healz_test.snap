
[TestHealzRoute - 1]
{"message":"OK"}
---
